lib/runtime-core/src/typed_func.rs

```diff
@@ -232,6 +281,7 @@ pub struct Func<'a, Args = (), Rets = (), Inner: Kind = Wasm> {
     inner: Inner,
     func: NonNull<vm::Func>,
     func_env: Option<NonNull<vm::FuncEnv>>,
+    signature: Option<Arc<FuncSig>>,
```

Comment:
Is it not possible to use the phantom data to get this information?  If so it would be good to leave a comment about why!  Seems like we were able to get by with never storing this information at runtime before